portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the value entered in the form.
+    String name = request.getParameter("name-input");
+    String email = request.getParameter("email-input");
+    String message = request.getParameter("message-input");
+    String wc = request.getParameter("wc-input");
+    String other = request.getParameter("other-input");
+    String language = request.getParameter("language");
+
+    // Print the value so you can see it in the server logs.
+    System.out.println("Name: " + name + "</p>");
```

Comment:
We're formatting the information twice here, which can lead to discrepancies if it changes between what gets printed in the logs and what gets sent back. This code would allow it to be formatted once and reduce the overall amount of code.

```
@Override
public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
    String name = request.getParameter("name");
    String email = request.getParameter("email");
    String subject = request.getParameter("subject");
    String message = request.getParameter("message");
    sendResponse(response, String.format("Name: %s", name));
    sendResponse(response, String.format("Email: %s", email));
    sendResponse(response, String.format("Subject: %s", subject));
    sendResponse(response, String.format("Message: %s", message));
}

private void sendResponse(HttpServletResponse response, string responseText) throws IOException {
    System.out.println(responseText);
    response.getWriter().println(responseText);
}
```

This also makes it so the logs only show up until the point that it stopped working, instead of logging everything all at once.